### DISCUSSION TITLE
Log files

### DISCUSSION THREAD

**Discussion by joelfiddes** [ArcticSnow/TopoPyScale#79] [Ideas]
log files would be useful for quick debugging but also mainly storing important info of a sim such as code version. In that way a sim would be perfectly reproducible some months down the line with the new requirements.yml file.

Currently I have situations were reprocucibility is challenging due to not knowing which version generated results

**Comment by ArcticSnow** [comment]
Yes, log file would be a neat feature! The [`logging` library](https://docs.python.org/3/howto/logging.html) should be quite straight forward. It also has logging levels with Warning, trace, info, error, etc. 

Also, for the version we could use the git commit hash: https://stackoverflow.com/questions/14989858/get-the-current-git-hash-in-a-python-script. This would help for us developing to know which commit version was used
